pkg/controller/networkpolicy/validate.go

```diff
@@ -680,9 +680,6 @@ func (v *antreaPolicyValidator) validatePeers(ingress, egress []crdv1beta1.Rule)
 	}
 	for _, rule := range egress {
 		if rule.ToServices != nil {
-			if !features.DefaultFeatureGate.Enabled(features.AntreaProxy) {
```

Comment:
No need to check antreaProxy.enable's value here?

Reply:
It will be checked on agent.

Reply:
There seems no check added in agent. I think we should fail the realization of a rule if it contains `toServices` but does't have AntreaProxy enabled. Otherwise there would be no issue when creating the rule but it would just not work without any error in API and log.

Reply:
https://github.com/antrea-io/antrea/pull/5401#discussion_r1330768457 I'll make another PR about this after merging this PR.

Reply:
Please create an issue in case it's forgotten.

Reply:
Added in #5591